simulation/traffic_simulator/src/data_type/lanelet_pose.cpp

```diff
@@ -76,6 +80,26 @@ auto CanonicalizedLaneletPose::canonicalize(
       ") is invalid, please check lanelet length, connection and entity route.");
   }
 }
+
+auto CanonicalizedLaneletPose::getAlternativeLaneletPoseBaseOnShortestRouteFrom(
+  LaneletPose from, const std::shared_ptr<hdmap_utils::HdMapUtils> & hdmap_utils) const
+  -> std::optional<LaneletPose>
+{
+  if (lanelet_poses_.empty()) {
+    return std::nullopt;
+  }
+  std::vector<std::int64_t> shortest_route =
+    hdmap_utils->getRoute(from.lanelet_id, lanelet_poses_[0].lanelet_id);
+  LaneletPose alternative_lanelet_pose = lanelet_poses_[0];
+  for (const auto laneletPose : lanelet_poses_) {
+    auto route = hdmap_utils->getRoute(from.lanelet_id, laneletPose.lanelet_id);
+    if (shortest_route.size() > route.size()) {
+      shortest_route = route;
+      alternative_lanelet_pose = laneletPose;
+    }
```

Comment:
```suggestion
  for (const auto & lanelet_pose : lanelet_poses_) {
    const auto route = hdmap_utils->getRoute(from.lanelet_id, lanelet_pose.lanelet_id);
    if (shortest_route.size() > route.size()) {
      shortest_route = route;
      alternative_lanelet_pose = lanelet_pose;
    }
  }
```

Reply:
Please add const and & if possible.

Reply:
I've applied the suggested change -> f0ef51bb33af1555161b5472b57285bdfec7a6ca